arbnode/espresso_state_checker.go

```diff
@@ -0,0 +1,100 @@
+package arbnode
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/offchainlabs/nitro/util/stopwaiter"
+	flag "github.com/spf13/pflag"
+)
+
+type StateCheckerConfig struct {
+	Enable          bool          `koanf:"enable"`
+	PollingInterval time.Duration `koanf:"polling-interval"`
+
+	// http endpoint of the trusted node
+	TrustedNodeUrl string `koanf:"trusted-node-url"`
+}
+
+var DefaultStateCheckerConfig = StateCheckerConfig{
+	Enable:          true,
+	PollingInterval: time.Second * 100,
+}
+
+func EspressoStateCheckerConfigAddOptions(prefix string, f *flag.FlagSet) {
+	f.Bool(prefix+".enable", DefaultStateCheckerConfig.Enable, "enable state checker")
```

Comment:
why do we have this flag? It should always be enabled?

Reply:
According to the design, this is an optional.

https://github.com/EspressoSystems/book/blob/49a3b57b26c59945b362781750a404f9c5e4f576/pages/nitro-caff-node-watchdog.tex#L148

Reply:
I discussed this with Jeb, this should not be optional and we should update the book and fix this